File: GO/CWE-798.go
package main

import (
	"fmt"
	"net/http"
)

func main() {
	// Жестко закодированные учетные данные
	username := "admin"
	password := "password123"

	http.HandleFunc("/login", func(w http.ResponseWriter, r *http.Request) {
		user := r.FormValue("username")
		pass := r.FormValue("password")

		if user == username && pass == password {
			fmt.Fprintf(w, "Вход выполнен успешно")
		} else {
			fmt.Fprintf(w, "Неверные учетные данные")
		}
	})

	fmt.Println("Сервер запускается на порту 8080...")
	http.ListenAndServe(":8080", nil)
}

File: С++/CWE-798.cpp
#include <iostream>
#include <string>

const std::string USERNAME = "admin";
const std::string PASSWORD = "password123";

bool authenticate(const std::string &username, const std::string &password) {
    if (username == USERNAME && password == PASSWORD) {
        return true; // Аутентификация успешна
    }
    return false; // Аутентификация не удалась
}

int main() {
    std::string username;
    std::string password;

    std::cout << "Введите имя пользователя: ";
    std::cin >> username;
    std::cout << "Введите пароль: ";
    std::cin >> password;

    if (authenticate(username, password)) {
        std::cout << "Доступ разрешен." << std::endl;
    } else {
        std::cout << "Доступ запрещен." << std::endl;
    }

    return 0;
}
